### PR TITLE
feat(es/minifier): ignore recursive usage for function and pure call

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**

This works suprisingly well for few lines of code

图片

And in the long run we could disable tsc minifier entirely as they would be mostly empty files.

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc_ecma_minifier/tests/fixture/issues/vercel/004/output.js [hunk 1]
```diff
@@ -1,9 +0,0 @@
-function ItemsList() {
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/fixture/issues/vercel/004/output.js:1]: This was a test about dependency issue of sequential inliner and I think we should not remove the test

**Review by kdy1 — CHANGES_REQUESTED** [review]
I'm not sure how should I review this PR properly 🤣

File: crates/swc_ecma_minifier/tests/fixture/issues/vercel/004/output.js (hunk 1 above)
Comment (Austaras) [reply crates/swc_ecma_minifier/tests/fixture/issues/vercel/004/output.js:1]: ~~It's not removed, it's just compressed~~

Should we add an export here?

Comment (kdy1) [reply crates/swc_ecma_minifier/tests/fixture/issues/vercel/004/output.js:1]: I think it will work

**Review by kdy1 — APPROVED** [review]
Thank you!


---

swc-bump:
 - swc_ecma_minifier

**Pull request merged**